### DISCUSSION TITLE
Inspecting bytecode of anonymous inner classes

### DISCUSSION THREAD

**Discussion by susliko** [scalameta/metals#4456] [Q&A]
## Intro
Some Scala constructs are implemented as anonymous [inner classes](https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.7.6) at the bytecode level, therefore a fraction of bytecode is hidden from view

#### Example 1
```scala
class A
trait B
class C {
  val ab = new A with B
}
```
Inspecting a bytecode of class `C` reveals  `ab` being an instance of such a anonymous inner class
```
...
InnerClasses:
  private static final #7;                // class C$$anon$1
```

#### Example 2
```scala
class A {
  val a = List().collect { case x => x }
}
```
Partial function inside `.collect` is represented as anonymous inner class as well:
```
InnerClasses:
  private static final #7;                // class A$$anon$1
```
## Question
1. Is there a functionality in metals, which would allow querying a bytecode for some class and all of its inner classes? If there is none, do you think there might be a straightforward implementation?
2. If such functionality were implemented, could we support it in nvim-metals' javap output view? E.g. just putting outputs for these classes one after another in the same buffer
```
Classfile A
<bytecode of A>

Classfile A$$anon$1
<bytecode of A$$anon$1>
```

Also, not sure if anyone else would be interested in this 😬

**Comment by ckipp01** [comment]
Hey @susliko thanks for the questions!

> Is there a functionality in metals, which would allow querying a bytecode for some class and all of its inner classes? If there is none, do you think there might be a straightforward implementation?

Currently, not that I know of. As far as implementing I'm actually not sure off the top of my head. However, this would be implemented in Metals itself, not the nvim extension, so I'll actually move this discussion over there to get some more insight on it from the others.

> If such functionality were implemented, could we support it in nvim-metals' javap output view? E.g. just putting outputs for these classes one after another in the same buffer

This _sort_ of depends on whether it's implemented as just another decoded option like show semanticdb, show javap, etc or if it would be implemented in a new way. So it'd be important to answer 1 first.
